src/main/java/eu/europeana/metis/sandbox/service/workflow/MediaProcessingServiceImpl.java

```diff
@@ -55,17 +54,20 @@ public RecordInfo processMedia(Record record) {
     var inputRdf = record.getContent();
     var rdfDeserializer = converterFactory.createRdfDeserializer();
 
-    // Get resource entries
-    var resourceEntries = getRdfResourceEntries(record, inputRdf, rdfDeserializer);
-
-    var rdfForEnrichment = getEnrichedRdf(record, inputRdf, rdfDeserializer);
-
     List<RecordError> recordErrors = new LinkedList<>();
+    EnrichedRdf rdfForEnrichment = getEnrichedRdf(record, inputRdf, rdfDeserializer);
+
     try (MediaExtractor extractor = processorFactory.createMediaExtractor()) {
-      for (RdfResourceEntry entry : resourceEntries) {
-        processResourceEntry(record, rdfForEnrichment, extractor, entry, recordErrors);
-      }
+      // Get main thumbnail
+      boolean hasThumbnail = processMainThumbnailResource(rdfDeserializer, inputRdf, extractor,
+          record, recordErrors);
+
+      // Get resource entries
+      processResourceEntry(record, rdfForEnrichment, extractor, inputRdf, hasThumbnail,
+          rdfDeserializer, recordErrors);
```

Comment:
Once you have solved the other two comments, you should find that the two methods you created are quite similar. Perhaps then you can merge them.

Reply:
Indeed, the methods became very similar. Created a new method that should work for both main thumbnail and remaining resources